Linter clean up

pablojimpas opened the pull request:
These are mostly small warnings that `go vet` or editor linters/analyzers show up, almost pedantic stuff but nice to have for long term maintenance.
There are a lot more fixes of this kind that can be made to other files, let me know if this PR is welcome and I will continue reviewing the rest of the codebase.

pablojimpas committed:
Fix linting problems in tasks

pablojimpas committed:
Fix linting problems in tasks

pablojimpas committed:
Fix linting problems in assets

pablojimpas committed:
Fix lint warnings in user

pablojimpas committed:
Merge commit '1bab159' into linter-clean-up

bepitulaz commented:
Hi @pablojimpas, thanks for the PR. Yes, please, you can continue reviewing the codebase.

Just take a note, you don't need to review the frontend code because I'm in the progress of rewriting it to React.

pablojimpas commented:
> Hi @pablojimpas, thanks for the PR. Yes, please, you can continue reviewing the codebase.
> 
> Just take a note, you don't need to review the frontend code because I'm in the progress of rewriting it to React.

Perfect, I'm only interested in the Go code for now.
Whenever I find time I will continue removing dead code and fixing these small details.

I will also try to address alerts from [LGTM](https://lgtm.com/projects/g/Tanibox/tania-core/). It will be cool to integrate a code analysis tool like this one in the repo.

Btw, looking forward to that frontend rewrite, I don't know if it's usual (since I try to avoid the JavaScript ecosystem as much as possible) but right now it's a dependency nightmare. Keep it simple :)

pablojimpas committed:
go fmt in tasks/query

pablojimpas committed:
Update dependencies and go mod maintenance

- very old version of satori/go.uuid with a known vulnerability

pablojimpas committed:
crypto/ssh/terminal deprecated in favour of x/term

pablojimpas committed:
main.go revision

pablojimpas committed:
Don't pass by value mock.Mock with a sync.Mutex

pablojimpas committed:
error strings should not be capitalized

pablojimpas committed:
using nicer language features

pablojimpas committed:
Unchecked errors and dead code

pablojimpas commented:
All `go vet` issues and linter warnings from text editors analyzers have been addressed. Feel free to review it and merge it if you like what you see.

Btw I've also updated module dependencies because some were really out of date or even unused, moreover the old version of satori/go.uuid had a security vulnerability.Btw I've also updated module dependencies because some were really out of date.

I leave [this](https://goreportcard.com/report/github.com/Tanibox/tania-core) here for reference, there is a lot of refactor that can be done to reduce cyclomatic complexity.

pablojimpas committed:
misspelled English words

pablojimpas committed:
Remove unnecessary import alias
